### PR TITLE
Use linter

### PR DESCRIPTION
**Pull request by radicaled**

### CODE REVIEW

File: lib/linter/linter_component.coffee [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+          messages.push
+            type: _.capitalize(error.severity.toLowerCase())
+            text: error.message
+            filePath: file
+            range: range
+
+      @linter.deleteProjectMessages(linterConfig)
+      @linter.setProjectMessages linterConfig, messages
+      @linter.views?.render()
```
Comment (steelbrain) [inline lib/linter/linter_component.coffee]: It is never null, so you can just drop the null-check

File: lib/linter/linter_component.coffee [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    @errorRepository.onChange _.debounce(updateErrors, 250, maxWait: 1000)
+
+linterConfig =
+  grammarScopes: ['source.dart']
+  scope: 'project'
+  lintOnFly: false
+  # lint: ->
```
Comment (steelbrain) [inline lib/linter/linter_component.coffee]: You can always return an empty array instead of commenting it out.

Comment (radicaled) [inline lib/linter/linter_component.coffee]: I don't think `lint` gets exercised for project scope files (based on a cursory look at the codebase), and we don't add the linter  via `addLinter()`. Is there any benefit to doing so?

Comment (steelbrain) [inline lib/linter/linter_component.coffee]: If you don't add it using `addLinter()` then it's perfectly fine to leave it like that.

**Comment by steelbrain** [comment]
Be advised that I've pushed the release where UI is automatically rendered when Project or EditorLinter messages are changed, so you can just drop that code in your package.

**Comment by radicaled** [comment]
Thanks mate.

**Pull request merged**